eos-phone-home

```diff
@@ -406,66 +403,73 @@ class PhoneHome(object):
             else:
                 # increment the counter upon successful ping only
                 count = self._lookup_or_get_variable('count')
                 self._set_count(count + 1)
 
-            print("Pinged!")
+            log.info("Pinged!")
         else:
-            print("Could not send ping to server!", file=sys.stderr)
+            log.warning("Could not send ping to server!")
 
             return False
 
         return True
 
-    def run(self):
+    def run(self, exit_on_server_error):
         if not os.path.isdir(self.STATE_DIRECTORY):
-            print("State folder (%s) not found! Exiting!" %
-                  self.STATE_DIRECTORY, file=sys.stderr)
+            log.critical("State folder (%s) not found! Exiting!",
+                         self.STATE_DIRECTORY)
             exit(1)
 
         if not os.access(self.STATE_DIRECTORY, os.W_OK):
-            print("State folder (%s) not writable! Exiting!" %
-                  self.STATE_DIRECTORY, file=sys.stderr)
+            log.critical("State folder (%s) not writable! Exiting!",
+                         self.STATE_DIRECTORY)
             exit(1)
 
         success = True
 
         try:
             if self._force or self._need_to_activate():
-                print("Activation needed.")
+                log.info("Activation needed.")
                 if not self._do_activate():
                     success = False
             else:
-                print("Already activated!")
+                log.info("Already activated!")
 
             if self._force or self._need_to_ping():
-                print("Ping needed.")
+                log.info("Ping needed.")
                 if not self._do_ping():
                     success = False
             else:
-                print("Ping not due yet.")
-        except Exception as e:
-            print("Unhandled exception:", e, file=sys.stderr)
-            success = False
+                log.info("Ping not due yet.")
+        except:
+            log.exception("Unhandled exception:")
+            exit(1)
 
-        if success:
-            exit(0)
-        else:
+        if not success and exit_on_server_error:
             exit(1)
 
-if __name__ == '__main__':
+
+def main():
     p = argparse.ArgumentParser()
     p.add_argument('--debug', action='store_true',
                    help='verbose output; disables any actual phoning home')
     p.add_argument('--force', action='store_true',
                    help='always collect data (implies --debug)')
+    p.add_argument('--exit-on-server-error', action='store_true',
+                   help='exit with a non-0 status if activation and/or ping '
+                        "can't be sent to the server")
 
     args = p.parse_args()
     is_debug = args.debug or args.force
+    logging.basicConfig(level=logging.DEBUG if is_debug else logging.INFO)
     if is_debug:
-        print("Debugging turned on!")
+        log.info("Debugging turned on!")
```

Comment:
You've got 2 if/thens here that practically do the same thing. Maybe doing something like this would be a bit more maintainable even if it's a bit less terse?:
```
logging_level = logging.INFO
if is_debug:
    log.info("Debugging turned on!")
    logging_level = logging.DEBUG

logging.basicConfig(level=logging_level)
```